aiida_nanotech_empa/workflows/gaussian/spin_workchain.py

```diff
@@ -0,0 +1,317 @@
+                   required=True,
+                   help='basis_set for scf')
+
+        spec.input('multiplicity_list',
+                   valid_type=List,
+                   required=True,
+                   help='spin multiplicities')
+
+        spec.outline(
+            cls.submit_opts, cls.inspect_opts, cls.submit_next_steps,
+            cls.inspect_next_steps,
+            if_(cls.is_gs_oss)(cls.submit_nat_orb,
+                               cls.inspect_nat_orb), cls.finalize)
+
+        spec.outputs.dynamic = True
+
+        spec.exit_code(
+            390,
+            "ERROR_TERMINATION",
+            message="One or more steps of the work chain failed.",
+        )
+
+    def submit_opts(self):
+
+        # multiplicity 0 means RKS calculation
+
+        for mult in self.inputs.multiplicity_list:
+
+            label = f"m{mult}_opt"
+
+            submitted_node = self.submit(
+                GaussianSpinOptWorkChain,
+                gaussian_code=self.inputs.gaussian_code,
+                formchk_code=self.inputs.formchk_code,
+                cubegen_code=self.inputs.cubegen_code,
+                structure=self.inputs.structure,
+                functional=self.inputs.functional,
+                basis_set_opt=self.inputs.basis_set_opt,
+                basis_set_scf=self.inputs.basis_set_scf,
+                multiplicity=Int(mult),
+            )
+            submitted_node.description = label
+            self.to_context(**{label: submitted_node})
+
+    def inspect_opts(self):
+
+        opt_energies = []
+
+        for mult in self.inputs.multiplicity_list:
+            label = f"m{mult}_opt"
+
+            # check if everything finished nicely
+            if not common.check_if_previous_calc_ok(self, self.ctx[label]):
+                return self.exit_codes.ERROR_TERMINATION
+
+            opt_energy = self.ctx[label].outputs.scf_energy
+            opt_energies.append(opt_energy)
+            self.out(f"m{mult}_opt_energy", opt_energy)
+            self.out(f"m{mult}_opt_structure",
+                     self.ctx[label].outputs.opt_structure)
+            self.out(f"m{mult}_opt_out_params",
+                     self.ctx[label].outputs.scf_out_params)
+            self.out(f"m{mult}_opt_cube_images",
+                     self.ctx[label].outputs.cube_image_folder)
+
+        gs_i = np.argmin(opt_energies)
+
+        # if open-shell singlet is degenerate with closed-shell solution, prefer closed-shell
+        if self.inputs.multiplicity_list[
+                gs_i] == 1 and 0 in self.inputs.multiplicity_list:
+            cs_i = self.inputs.multiplicity_list.index(0)
+            if np.abs(opt_energies[cs_i].value -
+                      opt_energies[gs_i].value) < 1e-6:
+                gs_i = cs_i
+
+        self.ctx.gs_mult = Int(self.inputs.multiplicity_list[gs_i]).store()
+        self.ctx.gs_energy = opt_energies[gs_i]
+        gs_opt_label = "m{}_opt".format(self.ctx.gs_mult.value)
```

Comment:
```suggestion
        gs_opt_label = f"m{self.ctx.gs_mult.value}_opt"
```
I think f-string is clearer here.